lib/utils/datasetState.hpp

```diff
@@ -805,4 +805,69 @@ class subfreqState : public datasetState {
     std::vector<uint32_t> _subfreqs;
 };
 
+/**
+ * @brief A dataset state that keeps the RFI frame-dropping information of a datatset.
+ *
+ * @author Rick Nitsche
+ */
+class RFIFrameDropState : public datasetState {
+public:
+    /**
+     * @brief Constructor
+     * @param data  The RFI frame-dropping information as serialized by
+     *              RFIFrameDropState::to_json().
+     */
+    RFIFrameDropState(const nlohmann::json& data) {
+        try {
+            enabled = data["enabled"].get<bool>();
+            thresholds = data["thresholds"].get<std::vector<std::pair<float, float>>>();
+        } catch (std::exception& e) {
+            throw std::runtime_error(
+                fmt::format(fmt("RFIFrameDropState: Failure parsing json data ({:s}): {:s}"),
+                            data.dump(4), e.what()));
+        }
+    }
+
+    /**
+     * @brief Constructor
+     * @param enabled       True, if RFI frame-dropping enabled
+     * @param thresholds    Vector of pairs: thresholds and fractions
+     */
+    RFIFrameDropState(bool enabled, std::vector<std::pair<float, float>> thresholds) :
+        enabled(enabled),
+        thresholds(thresholds) {}
+
+    /**
+     * @brief Get RFI frame-dropping enabled information.
+     *
+     * @return True if RFI frame-dropping enabled.
+     */
+    bool get_enabled() const {
+        return enabled;
+    }
+
+    /**
+     * @brief Get RFI frame-dropping thresholds (read only).
+     *
+     * @return Vector of pairs: Thresholds and fractions.
```

Comment:
You might want to even more clearly than this document that they are packed as `threshold, fraction` pairs, particularly for what's in the JSON serialisation.